pkgs/tools/package-management/poetry/plugins/poeblix.nix

```diff
@@ -16,20 +17,22 @@ buildPythonPackage rec {
     hash = "sha256-TKadEOk9kM3ZYsQmE2ftzjHNGNKI17p0biMr+Nskigs=";
   };
 
-  postPatch = ''
-    sed -i '/poetry =/d' pyproject.toml
-  '';
-
-  nativeBuildInputs = [
+  build-system = [
     poetry-core
   ];
 
-  doCheck = false;
+  buildInputs = [
+    poetry
+  ];
+
   pythonImportsCheck = ["poeblix"];
 
+  # Needs a working poetry with this plugin already installed
```

Comment:
But de do have a working poetry. The problem is rather pytest-pspec or so.

Reply:
You are right, pspec would be needed, also. But in a quick test locally, I bypassed pspec by executing pytest manually. And the tests start executing, but then fail, as they execute the plugin directly through poetry: https://github.com/spoorn/poeblix/blob/main/test/test_walkthrough.py#L10 , so they depend on having installed the plugin in poetry beforehand. This is also mentioned upstream at https://github.com/spoorn/poeblix/tree/main?tab=readme-ov-file#development

That's what I was referring to with needing the plugin installed in a poetry installation.

Apart from that, currently one of the tests, https://github.com/spoorn/poeblix/blob/main/test/test_walkthrough.py#L10 , fails with python 3.11 (in upstream, not related to nixpkgs).

I deactivated the tests in the initial PR as some of the other packaged poetry plugins also have them deactivated (due to other reasons). But I understand that this is not optimal, and if you prefer not having more plugins added without proper tests, then it might be better to go ahead with the revert. Please let me know.

Thank you for taking the time to look deeper into this.

Reply:
> they depend on having installed the plugin in poetry beforehand

There is no difference between that and having both poetry and poeblix in $PYTHONPATH.

> Apart from that, currently one of the tests, https://github.com/spoorn/poeblix/blob/main/test/test_walkthrough.py#L10 , fails with python 3.11 (in upstream, not related to nixpkgs).

Please open an upstream issue about that and quote it as the reason for disabling tests instead.

Reply:
@dotlambda I opened an upstream issue plus PR at https://github.com/spoorn/poeblix/issues/22 and reference it in the comment explaining why tests are disabled

Reply:
@dotlambda *Ping* :slightly_smiling_face: